graphql/schemabuilder/function.go

```diff
@@ -16,7 +16,11 @@ func (sb *schemaBuilder) buildFunction(typ reflect.Type, m *method) (*graphql.Fi
 	// If the method is federated, we want to built a graphql field that returns all
 	// fields on that object. This allows them to be sent as federated keys to other servers.
 	if m.FederationType != nil {
-		return sb.buildFederatedFunction(typ, m)
+		if m.IsShadow {
```

Comment:
Do you think we need to check if `typ` is `federation` where to make sure we only build the function to `federation` object?